src/porepy/numerics/mixed_dim/assembler.py

```diff
@@ -1202,3 +1204,121 @@ def num_dof(self):
+
+        names = []
+        for key in self.block_dof.keys():
+            names.append(key[1])
+
+        unique_vars = list(set(names))
+        s += f"Variable names {unique_vars}"
+
+        return s
+
+    def __repr__(self) -> str:
+        s = f"Assembler objcet with in total {self.num_dof()} dofs"
+        s += f" on {len(self.block_dof)} subdomain and interface variables.\n"
+
+        s += f"Maximum grid dimension: {self.gb.dim_max()}.\n"
+        s += f"Minimum grid dimension: {self.gb.dim_min()}.\n"
+
+        for dim in range(self.gb.dim_max(), self.gb.dim_min() - 1, -1):
+            s += f"In dimension {dim}: {len(self.gb.grids_of_dimension(dim))} grids.\n"
+            names = []
+            for key in self.block_dof.keys():
+                if isinstance(key[0], tuple):
+                    continue
+                elif key[0].dim == dim:
+                    names.append(key[1])
+
+            unique_vars = list(set(names))
+            s += f"All variables present in dimension {dim}: {unique_vars}\n"
+
+            # Also check if some subdomains of this dimension have a subset of the
+            # variables defined on the totality of the subdomains
+
+            # List of found special (subset) variable combinations
+            found_special_var_combination: List[Set[str]] = []
+            # Loop over all grids of this dimension
+            for g in self.gb.grids_of_dimension(dim):
+                # All variables on this subdomain
+                var = set(self.variables_of_grid(g))
+                # Check if this is a subset of the full variable list on this dimension
+                if len(var) != len(var.intersection(set(unique_vars))):
+
+                    # We will only report each subset variable definition once.
+                    # By default, assume this subset is new
+                    new_comb = True
+                    # Check if we have found it before
+                    for spec in found_special_var_combination:
+                        if len(spec) == len(spec.intersection(set(var))):
+                            new_comb = False
+                            break
+                    # if not bound before, register the new one, and report it
+                    if new_comb:
+                        found_special_var_combination.append(var)
+                        s += "Variable subset on at least one subdomain in "
+                        s += " dimension {dim}: {var}\n"
+
+        for dim in range(self.gb.dim_max(), self.gb.dim_min(), -1):
+            names = []
+            for g in self.gb.grids_of_dimension(dim):
+                for e, _ in self.gb.edges_of_node(g):
+                    if isinstance(key[0], tuple):
+                        names += self.variables_of_grid(e)
+            unique_vars = list(set(names))
+            s += f"All variables present on edges between dimensions {dim} and {dim-1}"
+            s += f" {unique_vars}"
```

Comment:
```suggestion
            s += f" {unique_vars}\n"
```

Reply:
if you do `assembler.__repr__()` for a gb with intersecting fractures, I think you will be missing a line separator.

Reply:
implemented in a more complete suggestion